omod/src/test/java/org/openmrs/module/bedmanagement/rest/resource/AdmissionLocationResourceTest.java

```diff
@@ -264,4 +302,38 @@ public void shouldGiveCorrectErrorWithNoAdmissionLocationTagDefined() throws Exc
 		// etc
 		handle(request);
 	}
+	
+	@Test
+	public void shouldSupportCustomRepresentation() throws Exception {
+		MockHttpServletRequest request = request(RequestMethod.GET, getURI() + "/" + getUuid());
+		request.setParameter("v", "custom:(ward:(name),bedLayouts:(bedNumber,bedTagMaps:(bedTag:(name))))");
+		SimpleObject object = deserialize(handle(request));
+		System.out.println(object);
+		Assert.assertEquals(2, object.keySet().size());
+		Map<String, Object> ward = object.get("ward");
+		Assert.assertNotNull(ward);
+		Assert.assertEquals(1, ward.keySet().size());
+		Assert.assertEquals("Orthopaedic ward", ward.get("name"));
+		List<Map<String, Object>> bedLayouts = object.get("bedLayouts");
+		Assert.assertNotNull(bedLayouts);
+		Assert.assertEquals(6, bedLayouts.size());
+		for (Map<String, Object> bedLayout : bedLayouts) {
+			Assert.assertEquals(2, bedLayout.size());
+			String bedNumber = (String)bedLayout.get("bedNumber");
+			Assert.assertNotNull(bedNumber);
+			if (bedNumber.equals("307-a")) {
+				List<Map<String, Object>> bedTagMaps = (List<Map<String, Object>>)bedLayout.get("bedTagMaps");
+				Assert.assertEquals(2, bedTagMaps.size());
+				for (Map<String, Object> bedTagMap : bedTagMaps) {
+					Assert.assertEquals(1, bedTagMap.size());
+					Map<String, Object> bedTag = (Map<String, Object>) bedTagMap.get("bedTag");
+					Assert.assertNotNull(bedTag);
+					Assert.assertEquals(1, bedTag.size());
+					String bedTagName = (String)bedTag.get("name");
+					Assert.assertNotNull(bedTagName);
+					Assert.assertTrue(bedTagName.equals("Oxygen") || bedTagName.equals("Broken"));
+				}
+			}
+		}
+	}
```

Comment:
All of these unit test additions are here to ensure that the previous json representations that were expected continue to be supported, and also that full and custom representations are supported.